gprofiler/main.py

```diff
@@ -523,13 +455,13 @@ def parse_cmd_args():
     if not args.upload_results and not args.output_dir:
         parser.error("Must pass at least one output method (--upload-results / --output-dir)")
 
-    if args.dwarf_stack_size > 65528:
+    if args.perf_dwarf_stack_size > 65528:
         parser.error("--perf-dwarf-stack-size maximum size is 65528")
 
     if args.perf_mode in ("dwarf", "smart") and args.frequency > 100:
         parser.error("--profiling-frequency|-f can't be larger than 100 when using --perf-mode 'smart' or 'dwarf'")
 
-    if args.nodejs_mode == "perf" and args.perf_mode not in ("fp", "smart"):
+    if args.perf_nodejs_mode == "perf" and args.perf_mode not in ("fp", "smart"):
```

Comment:
It is kinda weird, though, because `nodejs_mode` may contain values other than `perf`, in which case it might be totally unrelated to `perf`.

The correct way, maybe, is to define `nodejs_mode` and add an argument `perf_inject` which is a boolean set to `True` if `nodejs_mode == "perf"`.

Reply:
The only possible values are either `perf` or `disabled`. And in any case, I think the name is not horrible because behind the scenes the value is only relevant for the `perf` profiler.

Reply:
good